File: TFC-AppEventos/TFC.AppEventos.Application.DTO/UserDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TFC.AppEventos.Application.DTO
{
    public class UserDto
    {
        public int? UserId { get; set; }
        public int Name { get; set; }
        public int LastName { get; set; }
        public int Phone { get; set; }
        public DateOnly? BirthDate { get; set; }
        public int? City { get; set; }
        public int? Country { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string? Role { get; set; }
    }
}

File: Flutter/app_eventos/lib/core/models/dto/user_dto.dart
class UserDto {
  final int? userId;
  final String? name;
  final String? lastName;
  final int phone;
  final String? birthDate;
  final String? city;
  final String? country;
  final String? username;
  final String? email;
  final String? password;
  final String? role;

  UserDto({
    this.userId,
    required this.name,
    required this.lastName,
    required this.phone,
    this.birthDate,
    this.city,
    this.country,
    required this.username,
    required this.email,
    required this.password,
    this.role,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) => UserDto(
    userId: json['userId'],
    name: json['name'],
    lastName: json['lastName'],
    phone: json['phone'],
    birthDate: json['birthDate'],
    city: json['city'],
    country: json['country'],
    username: json['username'],
    email: json['email'],
    password: json['password'],
    role: json['role'],
  );

  Map<String, dynamic> toJson() => {
    'userId': userId,
    'name': name,
    'lastName': lastName,
    'phone': phone,
    'birthDate': birthDate,
    'city': city,
    'country': country,
    'username': username,
    'email': email,
    'password': password,
    'role': role,
  };
}